From da4d050014988c7e58da4e897aaf89e5f30cd613 Mon Sep 17 00:00:00 2001
From: Geoff Thorpe <Geoff.Thorpe@freescale.com>
Date: Wed, 28 Mar 2012 08:02:49 -0500
Subject: [PATCH 089/121] bman: fix return value for fsl_bpool_init()

In the absence of a "fsl,bpool-thresholds" property, this function was
inadvertently returning 4 rather than zero.

Signed-off-by: Geoff Thorpe <Geoff.Thorpe@freescale.com>
Acked-by: Bharat Bhushan <Bharat.Bhushan@freescale.com>
[Kevin: Original patch taken from FSL
QorIQ-SDK-V1.2-SOURCE-20120614-yocto.iso image]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/staging/fsl_qbman/bman_driver.c |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/staging/fsl_qbman/bman_driver.c b/drivers/staging/fsl_qbman/bman_driver.c
index 0d3734e..827666a 100644
--- a/drivers/staging/fsl_qbman/bman_driver.c
+++ b/drivers/staging/fsl_qbman/bman_driver.c
@@ -69,12 +69,13 @@ static int __init fsl_bpool_init(struct device_node *node)
 		if (ret)
 			pr_err("No CCSR node for %s property '%s'\n",
 				node->full_name, "fsl,bpool-thresholds");
+		return ret;
 #else
 		pr_err("Ignoring %s property '%s', no CCSR support\n",
 			node->full_name, "fsl,bpool-thresholds");
 #endif
 	}
-	return ret;
+	return 0;
 }
 
 static int __init fsl_bpid_range_init(struct device_node *node)
-- 
1.7.9.7

